caffe2/CMakeLists.txt

```diff
@@ -345,165 +345,9 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
-    ${TORCH_SRC_DIR}/csrc/jit/passes/utils/subgraph_utils.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/utils/check_alias_annotation.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/utils/memory_dag.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/prepack_folding.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/quantization.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/xnnpack_rewrite.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/fuse_linear.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/passes/freeze_module.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/print_handler.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/interface.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/cuda/interface.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_ops_utils.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_prim_ops.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_prim_ops_fulljit.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_prim_ops_c10.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_string_ops.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/register_special_ops.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/ir/scope.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/ir_emitter.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/name_mangler.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/testing/file_check.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/convert_to_ssa.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/exit_transforms.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/inline_loop_condition.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/schema_matching.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/script_type_parser.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/sugared_value.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/parser.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/builtin_functions.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/versioned_symbols.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/canonicalize_modified_loop.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/edit_distance.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/logging.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/api/module.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/api/object.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/jit_exception.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/string_to_type.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/serialization/source_range_serialization.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/serialization/type_name_uniquer.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/frontend/tracer.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/testing/hooks_for_testing.cpp
-    ${TORCH_SRC_DIR}/csrc/utils/tensor_flatten.cpp
-    ${TORCH_SRC_DIR}/csrc/utils/variadic.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/kernel_cache.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/compiler.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/executor.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/codegen.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/codegen/fuser/fallback.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/api/function_impl.cpp
-    ${TORCH_SRC_DIR}/csrc/jit/runtime/vararg_functions.cpp
-
-    ${TORCH_SRC_DIR}/csrc/jit/tensorexpr/bounds_inference.cpp
```

Comment:
@malfet do you have plan to read other lists (those gated by USE_DISTRIBUTED / USE_CUDA / etc) from build_variables.bzl as well?
I'm asking because I plan to remove 'tensorexpr/*' from mobile build to reduce binary size. My plan is to create a separate list in build_variables.bzl and move tensorexpr/* related stuff from "libtorch_cmake_sources" into it. I'd like to make sure it's aligned with your long term plan before making the change.

Reply:
Yes, that aligns with the longer term plan.
Please go ahead and create a list in build_variables.bzl and read it in caffe2/CmakeLists.txt using `get_filelist()` macro. And feel free to expand it to `append_filelist`, if it'll make more sense.